### PR TITLE
WSDL Support tool

### PR DESCRIPTION
**Pull request by gcornacchia**
These commits enable the tool to handle SOAP Service registration automation in API Manager,
The command line interface contains a new parameter to specify the URL of wsdl to import in API Manager

In the current version of API Manager REST API the import of SOAP Backend API is only supported with URL, not by file (as per REST API with swagger file)

### CODE REVIEW

**Comment by gcornacchia** [comment]
> @gcornacchia. I really appreciate the effort.
> Would it be possible to add a few integration tests for that? I would like to see the following scenarios covered:
> 
> * loading a WSDL from a URL (given the URL as you did)
> * loading a WSDL from Reference-File containing the URL to WSDL
> * loading a WSDL from a file location
> * Include for each a No-Change-Test (meaning, the tool must realize there is no change in the WSDL)
> 
> Integration tests that are similar:
> com.axway.apim.test.basic.NoChangeAPITestIT
> com.axway.apim.test.basic.SwaggerFromURLRefFileTestIT

Yes I will try to,
the third point I think is not applicable, as I understand from the documentation of the APIManager REST API is not possible to import the backend api from a wsdl file, but only from a URL.

If you know the right API method to invoke to do that, please let me know :)

**Comment by cwiechmann** [comment]
Yes, you are right. Forget about it. I was thinking to provide this option (from local-filesystem) to bypass the limitation in API-Manager, but it comes down that API-Manager must support it.

**Comment by gcornacchia** [comment]
I implemented 2 tests:
1)  one direct import of wsdl from url
2) one import of wsdl from url read from reference file.

In both cases the test fails on the second attempt of import (without any changes) but in some way it finds that the definitions are different:

Detected Swagger-Filesizes: API-Manager: 7884 vs. Import: 7880

When I build locally in my Windows environment all tests complete without errors.

There must be some kind of different configuration (maybe some charset encoding, I don't know), need further investigation

**Comment by cwiechmann** [comment]
Do you want me to have also a look into it, why it fails?

**Comment by gcornacchia** [comment]
> Do you want me to have also a look into it, why it fails?

Yes please, It would be great

**Comment by gcornacchia** [comment]
At line 93 of wsdl file there's a special char ©.

**Comment by cwiechmann** [comment]
Yeah, I have seen this and this is the cause. 
It seems, that the "Actual API" is wrong and the "Desired API" is correct. So, it means, it must be related getting the actual API (WSDL) from the API-Manager. The changes you made concerning the Content-Type are great and in the right direction. There one missing piece.

**Comment by cwiechmann** [comment]
I'm starting to think, that the API-Manager, running inside this Docker-Container, is delivering the WSDL in that way and it's not related to the code of Swagger-Promote.

**Comment by cwiechmann** [comment]
Confirmed, the behavior is because of the API-Manager running inside this Docker-Container and I guess, the default Character-Encoding is different, hence it fails.

**Comment by gcornacchia** [comment]
can you unlock the build https://travis-ci.org/Axway-API-Management-Plus/apimanager-swagger-promote/builds/514711808?

Then, how can we move forward?

**Comment by cwiechmann** [comment]
I'm checking, if I can adjust the behavior inside the Docker-Container and I will create a support-case to have this tracked internally at Axway.

File: src/test/java/com/axway/apim/test/basic/WSDLFromURLDirectTestIT.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+			.extractFromPayload("$.[?(@.path=='${apiPath}')].id", "apiId");
+		
+		echo("####### Re-Import API from URL without a change #######");
+		createVariable("wsdlURL", "http://www.dneonline.com/calculator.asmx?wsdl");
+		createVariable("configFile", "/com/axway/apim/test/files/basic/wsdl-minimal-config.json");
+		createVariable("status", "unpublished");
+		createVariable("expectedReturnCode", "10");
+		action(wsdlImport);
+		
+		echo("####### Setting the status to Published #######");
```
Comment (cwiechmann) [inline src/test/java/com/axway/apim/test/basic/WSDLFromURLDirectTestIT.java]: I have extended this test-case a bit to simulate a more realistic developer workflow.

**Comment by cwiechmann** [comment]
I have now updated the Docker-Image to use the right Character-Set and my Branch containing your changes is now running fine.
FYI: I have added the following jvm.xml into the API-Instance conf folder:
```xml
<ConfigurationFragment>
	<!-- Set the Default charset to UTF-8, otherwise existing WSDLs are exposed with the wrong encoding -->
	<VMArg name="-Dfile.encoding=UTF-8" />
</ConfigurationFragment>```

Additionally I have extended one of the test-cases in this PR to simulate a more real world API-Developer workflow. 

However, during manual tests I have seen, that 2 Backend-APIs are created based on the provided WSDL and I'm actually not sure, what happens on Re-Creation. Perhaps more and more BE-APIs are created, as the tool today is deleting only 1. I think this needs to be checked, perhaps covered in a test and improved.

And I was thinking do to some refactoring, once it's stable in the development branch:
Instead of having an additional parameter (-w) and I would make the -a (API-Definition) smart, to automatically detect what kind of API it is. With that, it doesn't complicate the CI/CD workflow, as it doesn't matter which API-Type is actually handled.  
I also believe it will simplify the code internally, as it seems that WSDL- and Swagger-Files can be handled internally almost in the same way, besides when it comes to the import.  
Hence I will refactor for instance the field: swaggerFile into something like apiDefinition to avoid such a miss leading messages:
Changed property: swaggerDefinition[.... 
or 
Detected Swagger-Filesizes: API-Manager: 7884 vs. Import: 7880

Hope that's okay with you and many thanks for your help.

**Comment by gcornacchia** [comment]
that's great, I agree with you!

**Pull request merged**